src/Compilers/CSharp/Test/Semantic/Semantics/TopLevelStatementsTests.cs

```diff
@@ -8907,10 +8936,553 @@ class Program
+
+            CompileAndVerify(comp, symbolValidator: validate, sourceSymbolValidator: validate);
+            comp.VerifyEmitDiagnostics();
+
+            void validate(ModuleSymbol module)
+            {
+                bool fromSource = module is SourceModuleSymbol;
+                var field = module.GlobalNamespace.GetMember<NamedTypeSymbol>("Program").GetField("<Property>k__BackingField");
+                Assert.False(field.ContainingType.IsImplicitlyDeclared);
+                var fieldAttributes = GetAttributeNames(field.GetAttributes().As<CSharpAttributeData>());
+                if (fromSource)
+                {
+                    Assert.Empty(fieldAttributes);
+
+                    Assert.Equal(new[] { "<top-level-statements-entry-point>", "System.Int32 Program.<Property>k__BackingField",
+                            "System.Int32 Program.Property { get; set; }", "System.Int32 Program.Property.get",
+                            "void Program.Property.set", "Program..ctor()" },
+                        field.ContainingType.GetMembers().ToTestDisplayStrings());
+                }
+                else
+                {
+                    Assert.Equal(new[] { "CompilerGeneratedAttribute", "DebuggerBrowsableAttribute" }, fieldAttributes);
+
+                    Assert.Equal(new[] { "System.Int32 Program.<Property>k__BackingField", "void Program.<Main>$(System.String[] args)",
+                            "System.Int32 Program.Property.get", "void Program.Property.set",
+                            "Program..ctor()", "System.Int32 Program.Property { get; set; }" },
+                        field.ContainingType.GetMembers().ToTestDisplayStrings());
+                }
+            }
+        }
+
+        [Fact]
+        public void SpeakableEntryPoint_XmlDoc()
+        {
+            var src = @"
+System.Console.Write(42);
+";
+
+            var comp = CreateCompilation(src, parseOptions: TestOptions.RegularWithDocumentationComments);
+            comp.VerifyDiagnostics();
+
+            var cMember = comp.GetMember<NamedTypeSymbol>("Program");
+            Assert.Equal("", cMember.GetDocumentationCommentXml());
+        }
+
+        [Fact]
+        public void SpeakableEntryPoint_XmlDoc_ProgramIsPartial()
+        {
+            var src = @"
+System.Console.Write(42);
+
+/// <summary>Summary</summary>
+public partial class Program { }
+";
+
+            var comp = CreateCompilation(src, parseOptions: TestOptions.RegularWithDocumentationComments);
+            comp.VerifyDiagnostics();
+
+            var cMember = comp.GetMember<NamedTypeSymbol>("Program");
+            Assert.Equal(
+@"<member name=""T:Program"">
+    <summary>Summary</summary>
+</member>
+", cMember.GetDocumentationCommentXml());
+        }
+
+        [Fact]
+        public void SpeakableEntryPoint_XmlDoc_ProgramIsPartial_NotCommented()
```

Comment:
nit: It feels like the motivating scenario was `typeof(Program).Assembly` and reflection. Consider testing this in a scenario where `partial class Program` is not explicitly declared.

Reply:
Good idea. Adding a typeof test.